docs/usage/configuration/config_documentation.md

```diff
@@ -436,6 +436,14 @@ Sub-options for each listener include:
 * `x_forwarded`: Only valid for an 'http' listener. Set to true to use the X-Forwarded-For header as the client IP. Useful when Synapse is
    behind a reverse-proxy.
 
+* `request_id_header`: The header extracted from each incoming request that is
```

Comment:
Should we cross reference this with the reverse proxy docs? Not sure if we do that with the `x_forwarded` option already.

Please add the version of Synapse that we added this option to!